cloud/scope/powervs_cluster.go

```diff
@@ -157,3 +261,1558 @@ func (s *PowerVSClusterScope) PatchObject() error {
 func (s *PowerVSClusterScope) Close() error {
 	return s.PatchObject()
 }
+
+// Name returns the CAPI cluster name.
+func (s *PowerVSClusterScope) Name() string {
+	return s.Cluster.Name
+}
+
+// Zone returns the cluster zone.
+func (s *PowerVSClusterScope) Zone() *string {
+	return s.IBMPowerVSCluster.Spec.Zone
```

Comment:
what happens when someone calls when this zone is null?!

Reply:
For zone and resource group we are adding webhook to make it mandatory for user to set when using create infra workflow which is identified with help of annotations. 
But even if we want to be safe then we need to add nil check under each caller of this function.

Reply:
its always safe to have null check :)

Reply:
Added check where ever the value is being referred.